### ISSUE TITLE
Negative value in Energy dashbord

### ISSUE THREAD

**Issue by malendure81** [home-assistant/frontend#17617]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Since 2023.8.1, energy dashbord show a negative value of production for my solar panel every morning at 2-3 am.
![image](https://github.com/home-assistant/frontend/assets/75225135/f09ef0fc-4bfd-40cf-8f26-24417f43b2d9)

this negative value are not showed in the sensor:
![image](https://github.com/home-assistant/frontend/assets/75225135/36b4d903-ca73-4f5d-9e55-9f28d3a4c3ee)
or even on the shelly 1pm module :
![image](https://github.com/home-assistant/frontend/assets/75225135/075e191c-1f85-407f-9b7a-68329fa951ad)

### Describe the behavior you expected

no negative should appear or not so high

### Steps to reproduce the issue

1. upgrade to 2023.8.1
2. consult energy dashboard

### What version of Home Assistant Core has the issue?

2023.8.1

### What was the last working version of Home Assistant Core?

2023.7.3

### In which browser are you experiencing the issue with?

google chrome Version 115.0.5790.173 or Home Assistant Mobile App

### Which operating system are you using to run this browser?

Windows 11 or Android 13

### State of relevant entities

```txt
state_class: total
unit_of_measurement: kWh
device_class: energy
icon: mdi:solar-power
friendly_name: total_day
```

**Comment by karwosts** [comment]
Either your state_class needs to be `total_increasing`, or `last_reset` is not being used correctly with a total sensor. See here:

https://developers.home-assistant.io/docs/core/entity/sensor/#state-class-total

When your daily total drops to 0 without a reset, a `total` class sensor is going to intrepret that as a massive instantaneous negative outflow, which you're seeing on the graph.

**Comment by malendure81** [comment]
Hi @karwosts,

Bo problem before with core 2023.7.3.

Yes I change the state_class after a rapid search on the net to find a solution.
I'll check tomorrow if the problem is solved or not.

**Comment by malendure81** [comment]
Solved with state class change and upgrade to 2023.8.2
